### ISSUE TITLE
esp32 mcp2515 not receiving data

### ISSUE THREAD

**Issue by aiotsi** [ttlappalainen/NMEA2000#440]
Hi. I have a problem with receiving data from one esp32 to another. I use simple temperature monitor to DataDisplay2 but nothing show in serial. When i use simple example of sending and receiving data send.ino and receiving.ino it's working but when i tried nmea2000 data to transfer no receiving anything. I connect my esp32 pins with mcp2515 board:
VCC -- ESP32 VIN
GND -- GND
CS -- D5
SO -- D19
SI -- D23
SCK -- D18
INT -- D4
Simple send.ino and receiving.ino work, but not nmea2000. I think it's something with code because simple send recive works. I have resisors jumper connected. If i tried ActisenseListener.ino and use Actisense NMEA Reader software there is always TimedOut. Thanks for help.
![listener](https://github.com/user-attachments/assets/042cd9ad-1813-43bb-aa04-dbf432d99cb1)

![mcp2515_CAN_bus](https://github.com/user-attachments/assets/d2fb5af0-e20d-49a6-a790-bad333a3f526)
[esp32.zip](https://github.com/user-attachments/files/18162312/esp32.zip)

**Comment by ttlappalainen** [comment]
Yes, for sure there is something in your code or hw - not in library.

1. If you are using ESP32, why do you use that MCP2515 module? ESP32 has internal CAN controller and needs only transceiver - I prefer isolated ISO1050. Note that your module is not isolated. You can read more information about its importance from library documentation. Usin MCP2525 module will just make things more complex.

2. Since you are using MCP2515 module, did you force compiler to use it? As default examples uses ESP32 internal CAN and you have not connected transceiver to it, so for sure it does not work. So as indocumentation you need to add several defines to yor project.
#define USE_N2K_CAN 1
#define N2k_SPI_CS_PIN GPIO_NUM_xx // Pin for SPI Can Select
#define N2k_CAN_INT_PIN GPIO_NUM_xx // Use interrupt  and it is connected to pin 21
#define USE_MCP_CAN_CLOCK_SET 8  // possible values 8 for 8Mhz and 16 for 16 Mhz clock
#include <NMEA2000_CAN.h>

See also:
- https://ttlappalainen.github.io/NMEA2000/page_h_w_set_up.html
- https://ttlappalainen.github.io/NMEA2000/_n_m_e_a2000___c_a_n_8h.html
